src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,187 @@
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.RemoveUnusedParametersAndValues;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnusedParametersAndValues
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.RemoveUnusedParametersAndValues), Shared]
+    internal class CSharpRemoveUnusedParametersAndValuesCodeFixProvider:
+        AbstractRemoveUnusedParametersAndValuesCodeFixProvider<ExpressionSyntax, StatementSyntax, BlockSyntax, 
+                                                               ExpressionStatementSyntax, LocalDeclarationStatementSyntax,
+                                                               VariableDeclaratorSyntax, ForEachStatementSyntax,
+                                                               SwitchSectionSyntax, SwitchLabelSyntax,
+                                                               CatchClauseSyntax, CatchClauseSyntax>
+    {
+        protected override BlockSyntax GenerateBlock(IEnumerable<StatementSyntax> statements)
+            => SyntaxFactory.Block(statements);
+
+        protected override SyntaxToken GetForEachStatementIdentifier(ForEachStatementSyntax node)
+            => node.Identifier;
+
+        protected override SyntaxNode UpdateNameForFlaggedNode(SyntaxNode node, SyntaxToken newName)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.IdentifierName:
+                    var identifierName = (IdentifierNameSyntax)node;
+                    return identifierName.WithIdentifier(newName.WithTriviaFrom(identifierName.Identifier));
+
+                case SyntaxKind.VariableDeclarator:
+                    var variableDeclarator = (VariableDeclaratorSyntax)node;
+                    return variableDeclarator.WithIdentifier(newName.WithTriviaFrom(variableDeclarator.Identifier));
+
+                case SyntaxKind.SingleVariableDesignation:
+                    return SyntaxFactory.SingleVariableDesignation(newName).WithTriviaFrom(node);
+
+                case SyntaxKind.CatchDeclaration:
+                    var catchDeclaration = (CatchDeclarationSyntax)node;
+                    return catchDeclaration.WithIdentifier(newName.WithTriviaFrom(catchDeclaration.Identifier));
+
+                default:
+                    throw ExceptionUtilities.Unreachable;
+            }
+        }
+
+        protected override ILocalSymbol GetSingleDeclaredLocal(LocalDeclarationStatementSyntax localDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(localDeclaration.Declaration.Variables.Count == 1);
```

Comment:
did something validate this earlier?

Reply:
Added a doc comment on the base method - this API is only called for local declaration statements generated by this fixer, which only declare a single local.

---
In reply to: [230603224](https://github.com/dotnet/roslyn/pull/30777#discussion_r230603224) [](ancestors = 230603224)